frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareFleetMaintained/FleetMaintainedAppDetailsPage/FleetMaintainedAppDetailsPage.tsx

```diff
@@ -196,7 +197,19 @@ const FleetMaintainedAppDetailsPage = ({
     } catch (error) {
       // quick exit if there was an error adding the software. Skip the policy
       // creation.
-      renderFlash("error", getErrorMessage(error));
+
+      const ae = (typeof error === "object" ? error : {}) as AxiosResponse;
+      if (
+        ae.status === 408 ||
+        getErrorMessage(error).includes("json decoder error") // 400 bad request when really slow
+      ) {
+        renderFlash(
+          "error",
+          "Couldn't upload. Request timeout. Please make sure your server and load balancer timeout is long enough."
+        );
+      } else {
+        renderFlash("error", "Couldn't add. Please try again.");
```

Comment:
So, this is hiding more detailed/potentially helpful error messages, which it seems like @ghernandez345 went out of his way to show in 631af6b8bfee3b5fe7becccc2fc53ceb1fddccbd. Gonna add him as a reviewer on this to triple-check but I want to say the fallback should be `renderFlash("error", getErrorMessage(error));` rather than hiding what the API is sending back.

Reply:
Yeah, looking through the code we seem to be very inconsistent with returning basic "Couldn't [perform action]." and actually surfacing the backend error. 

> None of these errors were coded to Figma spec, so I'm wondering if I'm missing a conversation around it.

Reply:
Alternatively, it could catch the request timeout if it 408 or has the json decoder error and surface `Couldn't upload. Request timeout. Please make sure your server and load balancer timeout is long enough.`, fall back to surface backend errors, and if that's missing fall back to surface `Couldn't add. Please try again.`

Reply:
That approach would make me feel better at least 😅

Reply:
Yes, this is removing some work I did for generating some other helpful UI errors. I would add back the `getErrorMessage` function and then extending that to return the new messages that you'd like to generate there. I wouldn't add it here as `getErrorMessage` is meant to encapsulate the complicated error message generating logic.